jstests/aggregation/isodates.js

```diff
@@ -0,0 +1,243 @@
+/*
```

Comment:
Ok points taken. It looks like this test is copied from testall.js, which is not my favorite test. It's a little hard to follow, uses bad variable names, different styles of comments, and relies on data that is provided from another file, which gives no intuition as to why the assertions might be true. Sorry for providing a bad example test. I'd like to eventually refactor and get rid of that test, but it's not very pressing...

I'd suggest rewriting this test to be more in the style of unit tests. [this test](https://github.com/mongodb/mongo/blob/r3.3.2/jstests/aggregation/bugs/server19548.js) comes to mind as a good example of providing good coverage, being easy to read, and having little test fixture overhead. I'll let you decide what's the best way to do this, but I'm thinking there's probably some to write the test such that we can provide one date and the expected values for $week, $isoWeek, $year, $isoWeekYear, $dayOfWeek, and $isoDayOfWeek.